src/datastore/pg-store.ts

```diff
@@ -4404,4 +4404,44 @@ export class PgStore extends BasePgStore {
     }
     return result;
   }
+
+  /** Retrieves the last transaction IDs with STX, FT and NFT activity for a principal */
+  async getPrincipalLastActivityTxIds(
+    principal: string
+  ): Promise<{ stx_tx_id: string | null; ft_tx_id: string | null; nft_tx_id: string | null }> {
+    const result = await this.sql<
+      { stx_tx_id: string | null; ft_tx_id: string | null; nft_tx_id: string | null }[]
+    >`
+      WITH last_stx AS (
+        SELECT tx_id
+        FROM principal_stx_txs
```

Comment:
I think this might not be considering `sponsor_address`. We'd need to update the `principal_stx_txs` table update logic for that. Thoughts?

Reply:
Good catch. Does being a sponsor generate an `stx_event`? If so I think we're covered. Otherwise we do need to patch the `principal_stx_txs` table.

Reply:
> Does being a sponsor generate an stx_event?

I don't think so